examples/logo_interpreter.rs

```diff
@@ -0,0 +1,86 @@
+//! To run a LOGO program, run:
+//!     cargo run --example logo_interpreter -- .\examples\my_logo_program.txt
+
+extern crate turtle;
+
+use turtle::Turtle;
+use std::env;
+use std::fs::File;
+use std::io::Read;
+use std::path::Path;
+
+fn main() {
+    let mut turtle = Turtle::new();
+    turtle.set_speed(4);
+
+    let args: Vec<String> = env::args().collect();
+    
+    if args.len() == 2 {
```

Comment:
Instead of reading the entire file here like this, could you read it line by line like in the example I provided you?

```rust
fn main() {
    let mut turtle = Turtle::new();

    let stdin = io::stdin();
    let mut reader = BufReader::new(stdin);
    loop {
        let mut buffer = String::new();
        let read_bytes = reader.read_line(&mut buffer)
            .expect("Unable read input from stdin");
        if read_bytes == 0 {
            // Reached EOF, stop interpreter
            break;
        }

        let mut cmd_args = buffer.split_whitespace().collect();
        handle_command(&mut turtle, &mut cmd_args);
    }
}
```

If you replace `io::stdin()` with `File::open(path).expect("Could not open provided program")`, you can use this code exactly as it is.

---

Reading the whole file at once might be useful in case someone tries to run code like this:
```
fd
27
```
However, I'd like to keep this simple for the moment and just support cases where people either put each command on separate lines or multiple *whole* commands on the same line.